tests/authentication/test_auth_change_password.py

```diff
@@ -28,41 +38,411 @@ class TestResetPassword:
+                generate_password(8),
+                400,
+                "Email must be valid",
+            ),
+            pytest.param(
+                "123-123-123",
+                "",
+                generate_password(8),
+                400,
+                "Email is required field",
+            ),
+        ],
+    )
+    def test_reset_password_incorrect_email(
+        self,
+        create_authorized_user,
+        code_for_reset_password,
+        email,
+        new_password,
+        expected_status_code,
+        expected_message_part,
+    ):
+        with step("Registration user"):
+            user = create_authorized_user["user"]
+
+        with step("Sent request to reset password"):
+            email = user["email"] if email is None else email
+            response_reset_password = AuthenticateAPI().change_password_through_reset(
+                email=email,
+                code_for_reset_password=code_for_reset_password,
+                new_password=new_password,
+                expected_status_code=400,
+            )
+
+        with step("Verify error message"):
+            assert_message_in_response(
+                response=response_reset_password, expected_message=expected_message_part
+            )
+
+        with step(
+            "Verify password does not reset successfully by Authorization request with new password"
+        ):
+            AuthenticateAPI().authentication(
+                email=email, password=new_password, expected_status_code=401
+            )
+
+    @pytest.mark.xfail(
+        reason="app accepted password that does not meet requirements."
+        "Bug: https://trello.com/c/ilaf1Luy/19-reset-password-the-app-allows-saving-passwords-that-do-not-meet-the-requirements"
+    )
+    @pytest.mark.critical
+    @severity(severity_level="CRITICAL")
+    @title("Test change password through reset password with incorrect password")
+    @description(
+        "Give user registered."
+        "WHEN user sent request to change password with incorrect password."
+        "THEN status HTTP CODE = 400 and appropriate error message"
+    )
     @pytest.mark.parametrize(
-        "register_user_and_reset_password",
+        "email_to_reset,code_for_reset_password, new_password,expected_status_code,expected_message_part",
         [
-            {
-                "firstName": firstName2,
-                "lastName": lastName2,
-                "password": password2,
-                "email_iced_late": email_iced_late,
-                "imap_server": imap_server2,
-                "email_address_to_connect": email_address_to_connect2,
-                "gmail_password": gmail_password2,
-            }
+            pytest.param(
+                None,
+                None,
+                generate_string(2),
+                400,
+                "Password must be at least 8 characters long and contain at least one letter, one digit, and may include special characters @$!%*?&",
```

Comment:
This message is repeated, i would extract it to a constant, e.g. `INCORRECT_PASSWORD_MSG=...`

Reply:
fixed